plans/025-clickhouse-error-sanitizer.md

```
@@ -0,0 +1,126 @@
+**Verify**: `cd apps/dashboard && bun run build` → exit 0.
+
+### Step 2: Unit-test the sanitizer
+
+Create `__tests__/sanitize-error.test.ts` (model after `error-response-builder.test.ts`). Cover: each bucket maps a representative raw message to its constant; an unknown message maps to the generic fallback; the output never contains the raw input (assert `expect(out).not.toContain(<a distinctive token from raw>)`).
+
+**Verify**: `cd apps/dashboard && bun test src/lib/api/error-handler` → all pass; `cd apps/dashboard && bun run type-check:test` → exit 0.
+
+### Step 3: Apply the sanitizer at the 3 `actions.ts` leak sites
+
+Replace the interpolated `error.message` in the client `message` with the sanitized value, keeping the action context prefix. Example for the kill-query site:
+```ts
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)